src/session.rs

```diff
@@ -108,11 +108,14 @@ impl Session {
                     .values()
                     .any(|user| !user.is_spectator && user.points.is_none())
                 {
-                    new_state.users.values_mut().for_each(|user| {
-                        if user.points.is_some() {
-                            user.points = Some("-1".to_string());
-                        }
-                    });
+                    new_state
+                        .users
+                        .iter_mut()
+                        .for_each(|(other_user_id, other_user)| {
```

Comment:
At this point the names `other_user_id` and `other_user` are not correct since `other_user_id` may be equal to `user_id`.

This can easily solved by adding a `filter` call directly after `iter_mut()`. Something along the lines of `.filter(|(user_id_ref, _)| *user_id_ref != user_id)`. Then the names would be correct and `other_user_id != user_id && other_user.points.is_some()`could be simplified to `other_users.points.is_some()`.

Reply:
The naming wasn't intended like that. It should just mean that it's _another variable_. Anyway, I tried to make it more concise as you suggested.